code/modules/unit_tests/species_unique_id.dm

```diff
@@ -0,0 +1,10 @@
+/datum/unit_test/species_unique_id
+
+/datum/unit_test/species_unique_id/Run()
+	var/list/gathered_species_ids = list()
+	for(var/datum/species/species as anything in subtypesof(/datum/species))
+		var/species_id = initial(species.id)
+		if(!(gathered_species_ids.Find(species_id)))
```

Comment:
Swap this, if-negative-else can be hard to read, especially when these are short